airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -293,6 +318,61 @@ private void reportFailure(final ConnectionUpdaterInput connectionUpdaterInput,
     }
   }
 
+  private SyncCheckConnectionFailure checkConnections(GenerateInputActivity.GeneratedJobInput jobInputs) {
+    final JobRunConfig jobRunConfig = jobInputs.getJobRunConfig();
+    final StandardSyncInput syncInput = jobInputs.getSyncInput();
+    final JsonNode sourceConfig = syncInput.getSourceConfiguration();
+    final JsonNode destinationConfig = syncInput.getDestinationConfiguration();
+    final IntegrationLauncherConfig sourceLauncherConfig = jobInputs.getSourceLauncherConfig();
+    final IntegrationLauncherConfig destinationLauncherConfig = jobInputs.getDestinationLauncherConfig();
+    final Long jobId = jobRunConfig.getJobId() != null ? Long.valueOf(jobRunConfig.getJobId()) : 1L;
+    final Integer attemptId = jobRunConfig.getAttemptId() != null ? Math.toIntExact(jobRunConfig.getAttemptId()) : 1;
+    SyncCheckConnectionFailure checkFailure = new SyncCheckConnectionFailure(jobId, attemptId);
+
+    final int attemptCreationVersion =
+        Workflow.getVersion(CHECK_BEFORE_SYNC_TAG, Workflow.DEFAULT_VERSION, CHECK_BEFORE_SYNC_CURRENT_VERSION);
+
+    if (attemptCreationVersion >= CHECK_BEFORE_SYNC_CURRENT_VERSION) {
+      final StandardCheckConnectionInput sourceConfiguration = new StandardCheckConnectionInput().withConnectionConfiguration(sourceConfig);
+      final CheckConnectionInput checkSourceInput = new CheckConnectionInput(jobRunConfig, sourceLauncherConfig, sourceConfiguration);
+      final String launchSourceDockerImage = sourceLauncherConfig.getDockerImage();
+
+      if (WorkerConstants.RESET_JOB_SOURCE_DOCKER_IMAGE_STUB.equals(launchSourceDockerImage) || checkFailure.isFailed()) {
+        log.info("SOURCE CHECK: Skipped");
+      } else {
+        log.info("SOURCE CHECK: Starting");
+        final StandardCheckConnectionOutput sourceCheckResponse = runMandatoryActivityWithOutput(checkActivity::run, checkSourceInput);
+        if (sourceCheckResponse != null && sourceCheckResponse.getStatus() == Status.FAILED) {
+          checkFailure.setFailureOrigin(FailureReason.FailureOrigin.SOURCE);
+          checkFailure.setFailureOutput(sourceCheckResponse);
+          log.info("SOURCE CHECK: Failed");
+        } else {
+          log.info("SOURCE CHECK: Successful");
+        }
+      }
+
+      final StandardCheckConnectionInput destinationConfiguration = new StandardCheckConnectionInput().withConnectionConfiguration(destinationConfig);
+      final CheckConnectionInput checkDestinationInput = new CheckConnectionInput(jobRunConfig, destinationLauncherConfig, destinationConfiguration);
+      final String launchDestinationDockerImage = destinationLauncherConfig.getDockerImage();
+
+      if (WorkerConstants.RESET_JOB_SOURCE_DOCKER_IMAGE_STUB.equals(launchDestinationDockerImage) || checkFailure.isFailed()) {
+        log.info("DESTINATION CHECK: Skipped");
```

Comment:
I think skipping the connection check for the source in the reset connection case makes sense, because when we reset a connection we just use a dummy empty source.

However, we may not actually want to skip the destination check for the reset connection case. In order to perform a reset successfully, I believe we need to be able to connect to the destination in order to clear it out.

Reply:
Agree! That's why this check was testing the destination's docker image, and we tested the source image above.  I wasn't sure if there was ever a time we replaced the destination connection like we do the source.  I'll change this to always run if the check hasn't yet failed, and check only the source with `workflowState.isResetConnection()`